### ISSUE TITLE
[BUG][JS] Description - ApiClient.js generated 'User-Agent' header but should be optional

### ISSUE THREAD

**Issue by dcmwong** [OpenAPITools/openapi-generator#12743]
##### Description

When using our site in Chrome there is an error in the console:
```
Refused to set unsafe header "User-Agent"
```
We use the openapi-generator-cli tool that generates the ApiClient.js setting the header `User-Agent` even though we haven't specified it in the cli.  There is no way, that we've seen, to tell the cli to not set the User-Agent header and therefore no way to remove the above error from Chrome, besides manually removing it after the build.

##### openapi-generator version

With version :
```
    "@openapitools/openapi-generator-cli": "^2.4.26",
```

The actual version in yarn is:
 "resolved" "https://registry.npmjs.org/@openapitools/openapi-generator-cli/-/openapi-generator-cli-2.4.26.tgz"

##### Generation Details

We use: this command to generate our js client code

```
openapi-generator-cli generate -g javascript -i swagger.json -o api --additional-properties=usePromises=true
```

##### Steps to reproduce

Generate client js code by running the below command on an example swagger.json
```
openapi-generator-cli generate -g javascript -i swagger.json -o api --additional-properties=usePromises=true
```

Navigate to the generated code `api/src/ApiClient.js` you will see the User-Agent code is there:

```
        /**
         * The default HTTP headers to be included for all API calls.
         * @type {Array.<String>}
         * @default {}
         */
        this.defaultHeaders = {
            'User-Agent': 'OpenAPI-Generator/1.0.0/Javascript'
        };

        /**

```
Even though it wasn't specified.

##### Related issues/PRs

I believe
[this commit](https://github.com/OpenAPITools/openapi-generator/commit/a127aab497807f2ba02f9eaed8a0e5f0182ff252#diff-873efc59f7090f596253e91897e6155e86b3fc75cebd2082ffff06010c794affR56) was where the issue was introduced.

##### Suggest a fix

The mustache template should probably check to see if the User Agent variable was passed before generating it.

**Comment by davesargrad** [comment]
I have this problem as well. This is really an issue for us.

**Comment by davesargrad** [comment]
This is easily solved at the application tier. Simply by adding the following line after construction of an instance:
```
let defaultClient = ApiClient.instance
defaultClient.defaultHeaders = {}
```

I've tested this, and it eliminates the chrome error.

**Comment by wing328** [comment]
closed via #20367 to add an option to skip setting the default header in ApiClient.js
